src/emitter.hpp

```diff
@@ -34,31 +34,117 @@ using std::thread;
 using std::condition_variable;
 using std::mutex;
 
+/**
+ * @brief Emitter is responsible for sending events to a Snowplow Collector.
+ * 
+ * Once the emitter receives an event from the Tracker a few things start to happen:
+ * 
+ * 1. The event is added to a local Sqlite3 database (blocking execution)
+ * 2. A long running daemon thread is started which will continue to send events as long as they can be found in the database (asynchronous)
+ * 3. The emitter loop will grab a range of events from the database up until the SendLimit
+ * 4. The emitter will send all of these events as determined by the Request, Protocol and ByteLimits
+ *    - Each request is sent in its thread.
+ * 5. Once sent it will process the results of all the requests sent and will remove all successfully sent events from the database
```

Comment:
That's a very nice touch!

Just a minor doubt about the wording.
I clearly see here the aim to provide info to help the user to make use of the Emitter in the proper way. I just see this a bit in the edge between providing useful info about emitter behaviour and explaining the internal implementation.

However, I'm not sure there are better ways to explain the behaviour of the Emitter.

Reply:
I understand the concern for explaining internal documentation. This explanation was actually taken from the official docs for the tracker (on docs.snowplowanalytics.com). My feeling is that the API docs can go into a bit more detail about the implementation of internal classes as they are updated together with them and bound together. Also there is some value to understanding how the Emitter works from the user perspective as they may be concerned about the blocking execution for example. But it's a fine line to walk between giving out the useful info and giving out too much.